src/libraries/System.Private.CoreLib/src/System/IO/BufferedStream.cs

```diff
@@ -184,12 +218,27 @@ public override long Length
                 EnsureNotClosed();
 
                 if (_writePos > 0)
-                    FlushWrite();
+                    FlushWrite(true);
 
                 return _stream!.Length;
             }
         }
 
+        // this method exists to keep old FileStream behaviour and don't perform a Flush when getting Length
```

Comment:
What is the "old" behavior it's keeping?  It'd be better to describe what it's achieving.

Reply:
https://github.com/dotnet/runtime/blob/832a65623d180412fcbaefd68aa57065341db931/src/libraries/System.Private.CoreLib/src/System/IO/FileStream.Windows.cs#L210-L212

Reply:
Can you keep that comment then?